exporter/elasticsearchexporter/config.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elasticsearchexporter
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/configtls"
+)
+
+// Config defines configuration for Elastic exporter.
+type Config struct {
+	configmodels.ExporterSettings `mapstructure:",squash"`
+
+	// URLs holds the Elasticsearch URLs the exporter should send events to.
+	//
+	// This setting is required if CloudID is not set and if the
+	// ELASTICSEARCH_URL environment variable is not set.
+	URLs []string `mapstructure:"urls"`
+
+	// CloudID holds the cloud ID to identify the Elastic Cloud cluster to send events to.
+	// https://www.elastic.co/guide/en/cloud/current/ec-cloud-id.html
+	//
+	// This setting is required if no URL is configured.
+	CloudID string `mapstructure:"cloudid"`
+
+	// Workers configures the number of workers publishing bulk requests.
+	Workers int `mapstructure:"workers"`
+
+	// Index configures the index, index alias, or data stream name events should be indexed in.
+	//
+	// https://www.elastic.co/guide/en/elasticsearch/reference/current/indices.html
+	// https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html
+	//
+	// This setting is required.
+	Index string `mapstructure:"index"`
+
+	// Pipeline configures the ingest node pipeline name that should be used to process the
+	// events.
+	//
+	// https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html
+	Pipeline string `mapstructure:"pipeline"`
+
+	HTTPClientSettings `mapstructure:",squash"`
+	Discovery          DiscoverySettings `mapstructure:"discover"`
+	Retry              RetrySettings     `mapstructure:"retry"`
```

Comment:
Consider using https://github.com/open-telemetry/opentelemetry-collector/blob/9de6dd77eeaef5fd2557a5080df14e9b083de938/exporter/exporterhelper/queued_retry.go#L58-L69

Reply:
Didn't uses the `RetrySettings` struct because retry logic on the HTTP level is actually provided by the go-elasticsearch library and Elasticsearch provides selective ACK which didn't seem to be a good fit for those helpers. As there is a chance that the `exporthelpers` do evolve/change (including `RetrySettings`) I'd rather not reuse it in order to not introduce setting/capabilities that do not fit the exporter.

Reply:
Alright, as long as you keep an eye on the names of the properties so that they follow the same name for the same things, I think it's OK